### ISSUE TITLE
`do list` does not work properly with path wildcards

### ISSUE THREAD

**Issue by TobiasNx** [metafacture/metafacture-fix#119]
[See here](https://metafacture.org/playground/?flux=PG_DATA%0A%7Cdecode-xml%0A%7Chandle-generic-xml%0A%7Cfix%0A%7Cencode-json%28prettyPrinting%3D%22true%22%29%0A%7Cprint%0A%3B%0A&fix=set_array%28%22coll%5B%5D%22%29%0Ado+list%28path%3A%22feld%3F%22%2C+%22var%22%3A%22%24i%22%29%0A++++add_field%28%22coll%5B%5D.%24append.feldtest%22%2C%22true%22%29%0A++++copy_field%28%22%24i.a.value%22%2C+%22coll%5B%5D.%24last.a%22%29%0Aend%0Aretain%28%22coll%5B%5D%22%29&data=%3C%3Fxml+version%3D%221.0%22+encoding%3D%22UTF-8%22%3F%3E%0A%3Crecord%3E%0A++++%3CfeldA%3E%3Ca%3EDog%3C/a%3E%3C/feldA%3E%0A++++%3CfeldA%3E%3Ca%3EApe%3C/a%3E%3C/feldA%3E%0A++++%3CfeldA%3E%3Ca%3EGiraffe%3C/a%3E%3C/feldA%3E%0A++++%3CfeldB%3E%3Ca%3ECrocodile%3C/a%3E%3C/feldB%3E%0A++++%3CfeldB%3E%3Ca%3EHuman%3C/a%3E%3C/feldB%3E%0A++++%3CfeldB%3E%3Ca%3EBird%3C/a%3E%3C/feldB%3E%0A%3C/record%3E&active-editor=fix)

Should be:
```
{
  "coll" : [ {
    "feldtest" : "true",
    "a" : "Dog"
  },{
    "feldtest" : "true",
    "a" : "Ape"
  },{
    "feldtest" : "true",
    "a" : "Giraffe"
  }, {
    "feldtest" : "true",
    "a" : "Crocodile"
  },
 {
    "feldtest" : "true",
    "a" : "Human"
  },
 {
    "feldtest" : "true",
    "a" : "Bird"
  }
 ]
}
```

**Comment by TobiasNx** [comment]
do list-Test: https://github.com/TobiasNx/fix-FunctionalReview-Testing/commit/67375bfae47e7aab2a4eda4b9ef2e5a029acbdfb

**Comment by blackwinter** [comment]
Turned example into unit test.

**Comment by fsteeg** [comment]
Functional review: @TobiasNx
Code review: @fsteeg

**Comment by TobiasNx** [comment]
This does also not work with pipes.

**Comment by TobiasNx** [comment]
With the new changes pipes do seem to work now. +1 For that . 

https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/5a064c36df1d4342446f0a01c2210089246abe44/data/testing/doList

But the changes in https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/5a064c36df1d4342446f0a01c2210089246abe44/data/approved/wildCardTest
are irritating.

**Comment by blackwinter** [comment]
> But the changes in [...] are irritating.

See discussion in #142.
